subsys/settings/src/settings_store.c

```diff
@@ -45,7 +45,7 @@ void settings_dst_register(struct settings_store *cs)
 
 static void settings_load_cb(char *name, char *val, void *cb_arg)
 {
-	settings_set_value(name, val);
+	ARG_UNUSED(settings_set_value(name, val));
```

Comment:
I'd rather see ARG_UNUSED only used for arguments, and not arbitrary expressions.